pricenode/src/main/java/bisq/price/spot/ExchangeRateProvider.java

```diff
@@ -60,4 +75,82 @@ protected void onRefresh() {
+        List<CurrencyPair> currencyPairs = exchange.getExchangeSymbols();
+
+        Set<String> supportedCryptoCurrencies = CurrencyUtil.getAllSortedCryptoCurrencies().stream()
+                .map(TradeCurrency::getCode)
+                .collect(Collectors.toSet());
+
+        Set<String> supportedFiatCurrencies = CurrencyUtil.getAllSortedFiatCurrencies().stream()
+                .map(TradeCurrency::getCode)
+                .collect(Collectors.toSet());
+
+        // Filter the supported fiat currencies (currency pair format is BTC-FIAT)
+        currencyPairs.stream()
+                .filter(cp -> cp.base.equals(Currency.BTC))
+                .filter(cp -> supportedFiatCurrencies.contains(cp.counter.getCurrencyCode()))
+                .forEach(cp -> {
+                    try {
+                        Ticker t = marketDataService.getTicker(new CurrencyPair(cp.base, cp.counter));
+
+                        result.add(new ExchangeRate(
+                                cp.counter.getCurrencyCode(),
+                                t.getLast(),
+                                // Some exchanges do not provide timestamps
+                                t.getTimestamp() == null ? new Date() : t.getTimestamp(),
+                                this.getName()
+                        ));
+                    } catch (CurrencyPairNotValidException cpnve) {
+                        // Some exchanges support certain currency pairs for other services but not for spot markets
+                        // In that case, trying to retrieve the market ticker for that pair may fail with this specific type of exception
+                        log.info("Currency pair " + cp + " not supported in Spot Markets: " + cpnve.getMessage());
+                    } catch (Exception e) {
+                        // Catch any other type of generic exception (IO, network level, rate limit reached, etc)
+                        log.info("Exception encountered while retrieving rate for currency pair " + cp + ": " + e.getMessage());
+                    }
+                });
+
+        // Filter the supported altcoins (currency pair format is ALT-BTC)
+        currencyPairs.stream()
+                .filter(cp -> cp.counter.equals(Currency.BTC))
+                .filter(cp -> supportedCryptoCurrencies.contains(cp.base.getCurrencyCode()))
+                .forEach(cp -> {
+                    try {
+                        Ticker t = marketDataService.getTicker(new CurrencyPair(cp.base, cp.counter));
+
+                        result.add(new ExchangeRate(
+                                cp.base.getCurrencyCode(),
+                                t.getLast(),
+                                // Some exchanges do not provide timestamps
+                                t.getTimestamp() == null ? new Date() : t.getTimestamp(),
+                                this.getName()
+                        ));
+                    } catch (CurrencyPairNotValidException cpnve) {
```

Comment:
Use `ex` for exception variable names. (`e` is also used widely throughout the codebase; `ex` is preferred).